framework/src/meshgenerators/StackGenerator.C

```diff
@@ -40,10 +40,18 @@ validParams<StackGenerator>()
   params.addParam<BoundaryName>("top_boundary", "top", "name of the top (y) boundary");
   params.addParam<BoundaryName>("bottom_boundary", "bottom", "name of the bottom (y) boundary");
 
+  // y boundary ids (2D case)
+  params.addParam<boundary_id_type>("top_boundary_id", "name of the top (y) boundary");
+  params.addParam<boundary_id_type>("bottom_boundary_id", "name of the bottom (y) boundary");
```

Comment:
I apologize for not noticing this before, but we never need to ask for either a boundary_(name) or boundary_id. As you can see it results in a lot of extra checking and it's a very common thing. In MOOSE we always just ask for "boundary" or "subdomain" and we use helper functions on MooseMesh to universally convert those to IDs in any case. So for instance, if a user supplies "boundary = top" or "boundary = 2" in their input file, the MooseMesh function will give you back "2" in both cases. I know you are on a very short time schedule, but if you have time, this could be simplified and cleaned up a bit more. Take a look at other objects that take boundaries (like BoundaryConditions) to see how it works.